packages/leap/lib/src/leap_map_transition_component.dart

```diff
@@ -0,0 +1,77 @@
+import 'dart:async';
+
+import 'package:flame/components.dart';
+import 'package:flame/effects.dart';
+import 'package:flutter/material.dart';
+import 'package:leap/leap.dart';
+
+abstract class LeapMapTransition extends PositionComponent {
+  LeapMapTransition({required this.game});
+
+  factory LeapMapTransition.defaultFactory(LeapGame game) {
+    return LeapFadeInOutMapTransition(
+      game: game,
+      duration: .4,
+      color: Colors.black,
+    );
+  }
+
+  final LeapGame game;
+
+  final _readyForNewMap = Completer<void>();
+  final _finished = Completer<void>();
+
+  void markReadyForNewMap() {
+    _readyForNewMap.complete();
+  }
+
+  void markFinished() {
+    _finished.complete();
+  }
+
+  Future<void> get newMapReadyToBeLoaded => _readyForNewMap.future;
+  Future<void> get finished => _finished.future;
+
+  void newMapLoaded();
+}
+
+class LeapFadeInOutMapTransition extends LeapMapTransition {
+  LeapFadeInOutMapTransition({
+    required super.game,
+    required this.duration,
+    required this.color,
+  });
+
+  final double duration;
+  final Color color;
+
+  @override
+  Future<void> onLoad() async {
+    await super.onLoad();
```

Comment:
It might be nice to do this in the base class like:

```dart

  @override
  Future<void> onLoad() async {
    await super.onLoad();
    intro();
  }

  void intro();
  void outro();
  
  
  Future<void> get introFinished => ...;
  Future<void> get outroFinished => ...;
```